Ref/Top/RefTopologyAc.cpp

```diff
@@ -0,0 +1,1336 @@
+      SG5.init(QueueSizes::SG5, InstanceIds::SG5);
+      blockDrv.init(QueueSizes::blockDrv);
+      chanTlm.init(QueueSizes::chanTlm, InstanceIds::chanTlm);
+      cmdDisp.init(QueueSizes::cmdDisp, InstanceIds::cmdDisp);
+      cmdSeq.init(QueueSizes::cmdSeq, InstanceIds::cmdSeq);
+      comm.init(InstanceIds::comm);
+      downlink.init(InstanceIds::downlink);
+      eventLogger.init(QueueSizes::eventLogger, InstanceIds::eventLogger);
+      fatalAdapter.init(InstanceIds::fatalAdapter);
+      fatalHandler.init(InstanceIds::fatalHandler);
+      fileDownlink.init(QueueSizes::fileDownlink, InstanceIds::fileDownlink);
+      fileManager.init(QueueSizes::fileManager, InstanceIds::fileManager);
+      fileUplink.init(QueueSizes::fileUplink, InstanceIds::fileUplink);
+      fileUplinkBufferManager.init(InstanceIds::fileUplinkBufferManager);
+      health.init(QueueSizes::health, InstanceIds::health);
+      linuxTime.init(InstanceIds::linuxTime);
+      pingRcvr.init(QueueSizes::pingRcvr, InstanceIds::pingRcvr);
+      prmDb.init(QueueSizes::prmDb, InstanceIds::prmDb);
+      rateGroup1Comp.init(QueueSizes::rateGroup1Comp, InstanceIds::rateGroup1Comp);
+      rateGroup2Comp.init(QueueSizes::rateGroup2Comp, InstanceIds::rateGroup2Comp);
+      rateGroup3Comp.init(QueueSizes::rateGroup3Comp, InstanceIds::rateGroup3Comp);
+      rateGroupDriverComp.init();
+      recvBuffComp.init();
+      sendBuffComp.init(QueueSizes::sendBuffComp, InstanceIds::sendBuffComp);
+      staticMemory.init(InstanceIds::staticMemory);
+      textLogger.init();
+      uplink.init(InstanceIds::uplink);
+    }
+
+    // Configure components
+    void configComponents(const TopologyState& state) {
+      cmdSeq.allocateBuffer(
+          0,
+          Allocation::mallocator,
+          ConfigConstants::cmdSeq::BUFFER_SIZE
+      );
+      downlink.setup(ConfigObjects::downlink::framing);
+      fileDownlink.configure(
+          ConfigConstants::fileDownlink::TIMEOUT,
+          ConfigConstants::fileDownlink::COOLDOWN,
+          ConfigConstants::fileDownlink::CYCLE_TIME,
+          ConfigConstants::fileDownlink::FILE_QUEUE_DEPTH
+      );
+      Svc::BufferManagerComponentImpl::BufferBins upBuffMgrBins;
+      memset(&upBuffMgrBins, 0, sizeof(upBuffMgrBins));
+      {
+        using namespace ConfigConstants::fileUplinkBufferManager;
+        upBuffMgrBins.bins[0].bufferSize = STORE_SIZE;
+        upBuffMgrBins.bins[0].numBuffers = QUEUE_SIZE;
+        fileUplinkBufferManager.setup(
+            MGR_ID,
+            0,
+            Allocation::mallocator,
+            upBuffMgrBins
+        );
+      }
+      health.setPingEntries(
+          ConfigObjects::health::pingEntries,
+          FW_NUM_ARRAY_ELEMENTS(ConfigObjects::health::pingEntries),
+          ConfigConstants::health::WATCHDOG_CODE
+      );
+      uplink.setup(ConfigObjects::uplink::deframing);
+    }
+
+    // Set component base Ids
+    void setBaseIds() {
+      blockDrv.setIdBase(0x100);
```

Comment:
@LeStarch I just realized that this code should use the symbolic base IDs, instead of the numeric ones. OK to push this change?

Reply:
@bocchino sure.

Reply:
Thanks, done.